### PR TITLE
Existing GA4 tag detected but useSnippet not disabled (3542).

### PR DESCRIPTION
**Pull request by johnPhillips**
## Summary

Addresses issue #3542

## Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: assets/js/modules/analytics/components/settings/SettingsForm.stories.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+export const WithGA4andUASnippet = Template.bind( null );
+WithGA4andUASnippet.storyName = 'UA and GA4 tag snippet switches';
+WithGA4andUASnippet.parameters = { features };
+
+export default {
+	title: 'Modules/Analytics/Settings/SettingsEdit',
+	decorators: [
+		( Story ) => {
+			const setupRegistry = ( registry ) => {
+				const accounts = fixtures.accountsPropertiesProfiles.accounts.slice( 0, 1 );
```
Comment (tofumatt) [inline assets/js/modules/analytics/components/settings/SettingsForm.stories.js:61]: What's wrong with:
```suggestion
				const accounts = fixtures.accountsPropertiesProfiles.accounts[ 0 ];
```
here? 🙂

File: assets/js/modules/analytics/components/common/UseUAandGA4SnippetSwitches.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+	if ( useUASnippet && useGA4Snippet ) {
+		message = 'Site Kit will add the Universal Analytics and Google Analytics 4 codes automatically';
+	} else if ( useUASnippet && ! useGA4Snippet ) {
+		message = 'Site Kit will add the Universal Analytics code automatically';
+	} else if ( ! useUASnippet && useGA4Snippet ) {
+		message = 'Site Kit will add the Google Analytics 4 code automatically';
+	} else {
+		message = 'Site Kit will not add the code to your site';
+	}
```
Comment (tofumatt) [inline assets/js/modules/analytics/components/common/UseUAandGA4SnippetSwitches.js:66]: These strings all need to be localisable:

```suggestion
	if ( useUASnippet && useGA4Snippet ) {
		message = __( 'Site Kit will add the Universal Analytics and Google Analytics 4 codes automatically', 'google-site-kit' );
	} else if ( useUASnippet && ! useGA4Snippet ) {
		message = __( 'Site Kit will add the Universal Analytics code automatically', 'google-site-kit' );
	} else if ( ! useUASnippet && useGA4Snippet ) {
		message = __( 'Site Kit will add the Google Analytics 4 code automatically', 'google-site-kit' );
	} else {
		message = __( 'Site Kit will not add the code to your site', 'google-site-kit' );
	}
```

File: assets/js/modules/analytics/components/common/UseUAandGA4SnippetSwitches.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+	const { setUseSnippet: setUseUASnippet } = useDispatch( STORE_NAME );
+	const { setUseSnippet: setUseGA4Snippet } = useDispatch( MODULES_ANALYTICS_4 );
+
+	const onUAChange = useCallback( () => {
+		setUseUASnippet( ! useUASnippet );
+		trackEvent( 'analytics_setup', useUASnippet ? 'analytics_tag_enabled' : 'analytics_tag_disabled' );
+	}, [ useUASnippet, setUseUASnippet ] );
+
+	const onGA4Change = useCallback( () => {
+		setUseGA4Snippet( ! useGA4Snippet );
```
Comment (tofumatt) [inline assets/js/modules/analytics/components/common/UseUAandGA4SnippetSwitches.js:49]: Looks like this is missing a `trackEvent` call, though I'll admit I'm not sure what it should be.

@felixarntz I assume we should add an event here, would:

```
trackEvent( 'analytics_setup', useUASnippet ? 'analytics4_tag_enabled' : 'analytics4_tag_disabled' );
```

make sense?

**Review by tofumatt — CHANGES_REQUESTED** [review]
Looks good! One change needed and one _very minor_ question for you in the test code.

The IB mentions `trackEvent` and I think we want one here, but I'll check with @felixarntz to be sure. After that this is good 👍🏻

File: assets/js/modules/analytics/components/common/UseUAandGA4SnippetSwitches.js (hunk 3 above)
Comment (felixarntz) [reply assets/js/modules/analytics/components/common/UseUAandGA4SnippetSwitches.js:49]: @tofumatt Good catch! Yes, let's use these event names you're suggesting. cc @johnPhillips

File: assets/js/modules/analytics/components/settings/SettingsForm.stories.js (hunk 1 above)
Comment (johnPhillips) [reply assets/js/modules/analytics/components/settings/SettingsForm.stories.js:61]: I actually shamelessly lifted this from elsewhere - I think the reasoning was that `slice` gives an array, which we then use later in a couple of places:
```js
const accountID = accounts[ 0 ].id;
// ...
registry.dispatch( STORE_NAME ).receiveGetAccounts( accounts );
```
I've refactored, I agree it seems a bit unwieldy here.

File: assets/js/modules/analytics/components/common/UseUAandGA4SnippetSwitches.js (hunk 3 above)
Comment (johnPhillips) [reply assets/js/modules/analytics/components/common/UseUAandGA4SnippetSwitches.js:49]: Thanks, I meant to ask about this. 👍

**Review by tofumatt — APPROVED** [review]
Looks great, thanks! 👍🏻

**Pull request merged**